### ISSUE TITLE
Compatibility with `numpy` 2.0

### ISSUE THREAD

**Issue by jdeschamps** [CAREamics/careamics#310]
## Problem

We capped the `numpy` dependency version to lower than 2.0. We should be compatible with latest versions.

**Comment by melisande-c** [comment]
Apparently there have been a lot of typing issues introduced recently (and we also saw this before we pinned the numpy version used by mypy in the pre-commit, see #341). See the following issues:

https://github.com/numpy/numpy/issues/28076
https://github.com/python/mypy/issues/18343

**Comment by tibuch** [comment]
Getting `careamics` compatible with `numpy>=2` would be great! Sooner or later this will lead to issues with pytorch and cuda drivers.

**Comment by jdeschamps** [comment]
Yep, to be perfectly honest, the only thing in the way is that we still have intel macOS laptops... (blah blah numpy 2.0 compatible with latest torch, latest torch not compatible with intel macOS).

This will happen in September!
